RE-SS3D/SS3D
# Apply Humble object pattern on substance containers

## Summary

Apply a soft version of the humble object pattern to allow tests on substance containers. This means well separate network/unity and substance container logic in methods. Also necessary in the context of my progresses on the health system. 

## Changes to Files

- Made cans variants of a generic can prefab, to allow for easy modifications on all cans. Necessary for testing reasons.
- Added some basic tests for substance containers.
- Fixed some encapsulation issues with stuff being exposed that should not be. 
- Expand documentation
- Added init method to substance container.

## Known issues

- You can currently transfer substances even if there's no room in the other container, is that a wanted behavior ?

To test the PR : 
- Connect with host and client.
- Spawn two cans.
- Try transferring some substances from one to another with client.
- Check in editor if that gets correctly updated on host side.
- Switch client and host in editor and redo that.
